pkgs/applications/misc/ianny/default.nix

```
@@ -0,0 +1,46 @@
+{ lib
+, fetchFromGitHub
+, rustPlatform
+, stdenv
+, dbus
+, pkg-config
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "ianny";
+  version = "0.1.0";
+
+  src = fetchFromGitHub {
+    owner = "zefr0x";
+    repo = "ianny";
+    rev = "c0e519b3258e837184e639eb9174394be700da7f";
+    hash = "sha256-+z8WXk8x/LHADzhOH6JlSuKDRhNvwrUTyTGcIfHB82k=";
+  };
+
+  cargoHash = "sha256-ieUCWgRumwAW6va8J7OxlQYvQsqGqxYoS9XHpXx++2A=";
+
+  nativeBuildInputs = [ pkg-config ];
+  buildInputs = [ dbus.dev ];
+
+  # Patch to fix the /usr/ paths hardcoded within the main.rs file to point to the correct location for the program to operate fully
+  patches = [ ./fix-paths.patch ];
+
+  postPatch = ''
+    substituteInPlace src/main.rs --replace '@out@' $out
```

```suggestion
    substituteInPlace src/main.rs \
      --replace '/usr/share/locale' $out/share/locale
```
with this we can avoid the patch file